### PR TITLE
Make Form Steps Look All Purdy

### PR DESCRIPTION
**Pull request by tn3rb**
gosh dang i just hads ta make dem form steps go n look all purdy like.

ScreenShot_20200512185647

now hope y'alls can scuse me cuz i gots ta go brush da poppy cone out from between my toofers!

### CODE REVIEW

File: assets/src/domain/shared/entities/datetimes/predicates/isSoldOut.ts [hunk 1]
```diff
@@ -1,7 +1,10 @@
-import capacityAtOrAbove from './capacityAtOrAbove';
 import { Datetime } from '@edtrServices/apollo/types';
 import { isBooleanTrue } from '@appServices/utilities';
 
-const isSoldOut = (date: Datetime): boolean => isBooleanTrue(date.isSoldOut) || capacityAtOrAbove(date, 100);
+const isSoldOut = (date: Datetime): boolean => isBooleanTrue(date.isSoldOut) || (
+	isFinite(date.capacity)
+	&& date.capacity > -1
+	&& date.capacity <= date.sold
+);
```
Comment (manzoorwanijk) [inline assets/src/domain/shared/entities/datetimes/predicates/isSoldOut.ts:8]: Prettier is not happy with this.

File: assets/src/domain/shared/entities/tickets/predicates/isSoldOut/index.ts [hunk 2]
```diff
@@ -1,7 +1,10 @@
 import { isBooleanTrue } from '@appServices/utilities';
 import { Ticket } from '@edtrServices/apollo/types';
 
-const isSoldOut = (ticket: Ticket): boolean =>
-	isBooleanTrue(ticket.isSoldOut) || (isFinite(ticket.quantity) && ticket.quantity <= ticket.sold);
+const isSoldOut = (ticket: Ticket): boolean => isBooleanTrue(ticket.isSoldOut) || (
+	isFinite(ticket.quantity)
+	&& ticket.quantity > -1
+	&& ticket.quantity <= ticket.sold
+);
```
Comment (manzoorwanijk) [inline assets/src/domain/shared/entities/tickets/predicates/isSoldOut/index.ts:8]: This as well

**Comment by nicolad** [comment]
Looks great:

image
***
one optional suggestion is to make active step a bit more noticeable like:

image

File: assets/src/infrastructure/ui/display/steps/Step.tsx [hunk 3]
```diff
@@ -3,17 +3,37 @@ import classNames from 'classnames';
...
+const Step: React.FC<StepProps> = ({
+	active,
+	className,
+	description = '',
+	disabled,
+	icon: Icon,
+	showStepNumber,
+	stepNumber,
+	title,
+	vertical = false
```
Comment (nicolad) [inline assets/src/infrastructure/ui/display/steps/Step.tsx:15]: :tada:

File: assets/src/infrastructure/ui/display/steps/Step.tsx [hunk 4]
```diff
@@ -3,17 +3,37 @@ import classNames from 'classnames';
...
+}) => {
+	const newClassName = classNames(
```
Comment (nicolad) [inline assets/src/infrastructure/ui/display/steps/Step.tsx:17]: if we'd go this way, than `enhancedClassName` might be a better name, `newClassName` isn't describing well what it is actually doing

File: assets/src/infrastructure/ui/display/steps/Step.tsx [hunk 5]
```diff
@@ -3,17 +3,37 @@ import classNames from 'classnames';
...
-			{Icon && <Icon />}
-			{showStepNumber && stepNumber}{' '}
-			{title}
-		</div>
+		<li className={ newClassName }>
+			{ showStepNumber && <div className={ 'ee-form-step__number' }>{ stepNumber }</div> }
+			{ Icon && <div className={ 'ee-form-step__icon' }><Icon /></div> }
+			<div className={ 'ee-form-step__content' }>
+				{ title && <div className={ 'ee-form-step__title' }>{ title }</div> }
+				{ description && <div className={ 'ee-form-step__desc' }>{ description }</div> }
```
Comment (nicolad) [inline assets/src/infrastructure/ui/display/steps/Step.tsx:34]: good naming of nested blocks :+1:

File: assets/src/infrastructure/ui/display/steps/Steps.tsx [hunk 6]
```diff
@@ -1,23 +1,39 @@
...
+const Steps: React.FC<StepsProps> = ({
+	children,
+	current = 0,
+	initial = 0,
+	showStepNumber,
+	vertical = false,
+	...props
+}) => {
+	const wrapperClassName = classNames('ee-form-steps__wrapper', props.className);
+	const listClassName = classNames('ee-form-steps', {
```
Comment (nicolad) [inline assets/src/infrastructure/ui/display/steps/Steps.tsx:17]: nice separation of class names :+1:

File: assets/src/infrastructure/ui/display/steps/Steps.tsx [hunk 7]
```diff
@@ -1,23 +1,39 @@
...
+		'ee-form-steps--vertical': vertical,
+		'ee-form-steps--inline': !vertical,
+	});
+	const heading = props.heading ?? __('Steps');
```
Comment (nicolad) [inline assets/src/infrastructure/ui/display/steps/Steps.tsx:21]: :+1:

**Pull request merged**